metals/src/main/scala/scala/meta/internal/metals/MetalsLanguageServer.scala

```diff
@@ -1967,6 +1970,70 @@ class MetalsLanguageServer(
       case ServerCommands.NewScalaProject() =>
         newProjectProvider.createNewProjectFromTemplate().asJavaObject
 
+      case ServerCommands.SetupNotebookKernelForThisProject() =>
```

Comment:
We could probably do this automatically if a notebook path is sent.

Reply:
Would you have a hint, where to look in metals, for when such a path might be sent?

Reply:
That's the method annotated with `@JsonNotification("textDocument/didOpen")` it's send every time a file is open.

https://github.com/scalameta/metals/blob/main/metals/src/main/scala/scala/meta/internal/metals/MetalsLanguageServer.scala#L1066